src/services/photos/database/photosRealmDB.ts

```diff
@@ -0,0 +1,170 @@
+  }
+
+  clear() {
+    if (!this._realm) return;
+    this.realm.write(() => {
+      this.realm.deleteAll();
+    });
+
+    fs.unlinkIfExists(this.realm.path);
+  }
+
+  private get realm() {
+    if (!this._realm) throw new Error('No connection open, call init() first');
+
+    return this._realm as Realm;
+  }
+
+  public async saveDevicePhotos(devicePhotos: Asset[]) {
+    this.realm.write(() => {
+      devicePhotos.forEach((devicePhoto) => {
+        this.realm.create<DevicePhotoDoc>(
+          DevicePhotoSchema.name,
+          {
+            name: devicePhoto.filename,
+            devicePhotoJSON: JSON.stringify(devicePhoto),
+            takenAt: new Date(devicePhoto.creationTime),
+          },
+          Realm.UpdateMode.All,
+        );
+      });
+    });
+  }
+
+  public async savePhotosItem(photo: Photo, getQueryTime = false) {
+    const start = Date.now();
+    this.realm.write(() => {
+      this.realm.create<PhotoDoc>(
+        PhotoSchema.name,
+        {
+          photoId: photo.id,
+          hash: photo.hash,
+          takenAt: photo.takenAt,
+          name: photo.name,
+          photoJSON: JSON.stringify(photo),
+          status: photo.status,
+        },
+        Realm.UpdateMode.All,
+      );
+    });
+
+    if (getQueryTime) {
+      this.logger.info(`Photos item saved into DB in ${Date.now() - start}ms`);
+    } else {
+      this.logger.info('Photos item saved into DB');
+    }
+  }
+
+  public async getSyncedPhotoByNameAndDate(name: string, takenAt: number) {
+    const docs = this.realm
+      .objects<PhotoDoc>(PhotoSchema.name)
+      .filtered('name == $0 AND takenAt == $1', name, new Date(takenAt));
+
+    return this.parseFirst(docs);
+  }
+
+  public async getSyncedPhotosCount() {
+    return this.realm.objects<PhotoDoc>(PhotoSchema.name).length;
+  }
+
+  public async getSyncedPhotoByName(name: string) {
+    const docs = this.realm.objects<PhotoDoc>(PhotoSchema.name).filtered('name == $0', name);
+
+    return this.parseFirst(docs);
+  }
+
+  public async getSyncedPhotoByHash(hash: string) {
+    const docs = this.realm.objects<PhotoDoc>(PhotoSchema.name).filtered('hash == $0', hash);
+
+    return this.parseFirst(docs);
+  }
+
+  public async deleteSyncedPhotosItem(photoId: string) {
+    const docs = this.realm.objects<PhotoDoc>(PhotoSchema.name).filtered('photoId == $0', photoId);
+    if (!docs[0]) return false;
+
+    this.realm.write(() => {
+      docs[0].status = PhotoStatus.Deleted;
+    });
+
+    return true;
+  }
+
+  public async getSyncedPhotos() {
+    const docs = this.realm.objects<PhotoDoc>(PhotoSchema.name);
+
+    return docs.map(this.parseObject);
+  }
+
+  private parseObject(object: Realm.Object) {
+    const serialized = object.toJSON() as PhotoDoc;
+    return JSON.parse(serialized.photoJSON) as Photo;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private parseFirst(objects: Realm.Results<any>) {
```

Comment:
using `unknown` will be more semantic and may allow you to remove the `eslint-disable` line